controllers/dscinitialization/servicemesh_setup.go

```diff
@@ -86,6 +86,47 @@ func configureServiceMeshFeatures() feature.FeaturesProvider {
 			}
 		}
 
+		cfgMapErr := feature.CreateFeature("shared-config-maps").
+			For(handler).
+			WithResources(servicemesh.ConfigMaps).
+			Load()
+		if cfgMapErr != nil {
+			return cfgMapErr
+		}
+
+		extAuthzErr := feature.CreateFeature("service-mesh-control-plane-setup-external-authorization").
+			For(handler).
+			Manifests(
+				path.Join(feature.AuthDir, "auth-smm.tmpl"),
+				path.Join(feature.AuthDir, "base"),
+				path.Join(feature.AuthDir, "mesh-authz-ext-provider.patch.tmpl"),
+			).
+			WithData(servicemesh.ClusterDetails).
+			PreConditions(
+				feature.EnsureCRDIsInstalled("authconfigs.authorino.kuadrant.io"),
+				servicemesh.EnsureServiceMeshInstalled,
+				servicemesh.EnsureAuthNamespaceExists,
+			).
+			PostConditions(
+				feature.WaitForPodsToBeReady(serviceMeshSpec.ControlPlane.Namespace),
+				func(f *feature.Feature) error {
+					return feature.WaitForPodsToBeReady(f.Spec.Auth.Namespace)(f)
+				},
+				func(f *feature.Feature) error {
+					// We do not have the control over deployment resource creation.
+					// It is created by Authorino operator using Authorino CR
+					//
+					// To make it part of Service Mesh we have to patch it with injection
+					// enabled instead, otherwise it will not have proxy pod injected.
+					return f.ApplyManifest(path.Join(feature.AuthDir, "deployment.injection.patch.tmpl"))
```

Comment:
I think it can be debatable how to name it.
That said, I do agree that this is not really a postcondition.

I see this more like a workaround, given `Features`  currently don't have a way to do processing after applying manifests.

Reply:
sure, not saying we need to rename what we have , just food for thought, maybe we need to extend a bit in the future when we need more support in similar case